File: src/main/java/com/matamercer/microblog/security/jwt/JwtConfig.java
package com.matamercer.microblog.security.jwt;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.http.HttpHeaders;
import org.springframework.stereotype.Component;

@Getter
@Setter
@NoArgsConstructor
@Component
@ConfigurationProperties(prefix = "application.jwt")
public class JwtConfig {
    private String secretKey;
    private String tokenPrefix;
    private Integer refreshTokenExpirationInDays;
    private Integer accessTokenExpirationInHours;

    public String getAuthorizationHeader(){
        return HttpHeaders.AUTHORIZATION;
    }

    public String getRefreshTokenHeader(){
        return "refreshToken";
    }




}

File: src/main/java/com/matamercer/microblog/security/authentication/JwtConfig.kt
package com.matamercer.microblog.security.authentication

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.http.HttpHeaders
import org.springframework.stereotype.Component

@Component
@ConfigurationProperties(prefix = "application.jwt")
class JwtConfig {
    var secretKey: String? = null
    var tokenPrefix: String = "Bearer"
    var refreshTokenExpirationInDays: Int = 0
    var accessTokenExpirationInHours: Int = 0
    val authorizationHeader: String
        get() = HttpHeaders.AUTHORIZATION
    val refreshTokenHeader: String
        get() = "refreshToken"
}